src/makeCalculator.js

```diff
@@ -5,6 +5,47 @@
  */
 function makeCalculator() {
   // write code here
+  const calculator = {
+    result: 0,
+
+    operate(operation, number = 0) {
+      operation.call(this, number);
+
+      return this;
+    },
+
+    add(number) {
+      this.result += number;
+
+      return this;
+    },
+
+    subtract(number) {
+      this.result -= number;
+
+      return this;
+    },
+
+    multiply(number) {
+      this.result *= number;
+
+      return this;
+    },
+
+    divide(number) {
+      this.result /= number;
+
+      return this;
+    },
+
+    reset() {
+      this.result = 0;
+
+      return this;
```

Comment:
The `reset` method properly sets the `result` property to 0 and allows for chaining by returning `this`. This is in line with the task requirements.